apps/wing-console/packages/use-persistent-state/src/index.tsx

```diff
@@ -0,0 +1,66 @@
+import {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  createContext,
+  useContext,
+  useEffect,
+  useRef,
+  MutableRefObject,
+  useState,
+} from "react";
+
+const PersistentStateContext = createContext<{
+  state: MutableRefObject<Map<string, any[]>>;
+}>(undefined!);
```

Comment:
If we do `undefined!`, at some point we may end up trying to use `createPersistentState` without using the `PersistentStateProvider`, and it will fail silently.

It's better to:

```suggestion
const PersistentStateContext = createContext<{
  state: MutableRefObject<Map<string, any[]>>;
} | undefined>(undefined);
```

And create an auxiliary `usePersistentState` method that throws if the context is undefined.